New article: 40x Faster! We rewrote our project with Rust

Peefy opened the pull request:
40x Faster! We rewrote our project with Rust
Two articles in English and Chinese.

Folyd commented:
Thanks for the contribution. Don't forget to update the `zine.toml` like this:
![img_v2_7ec9baa0-5f60-48d2-abc6-e3c25062076g](https://user-images.githubusercontent.com/3369694/215756243-279cc5c7-fd6a-409a-bb8c-350918218f7c.jpg)

BTW, you can use `zine` to preview the magazine in advance on your local computer, see https://rustmagazine.org/contribution/

Peefy commented:
> Thanks for the contribution. Don't forget to update the `zine.toml` like this: ![img_v2_7ec9baa0-5f60-48d2-abc6-e3c25062076g](https://user-images.githubusercontent.com/3369694/215756243-279cc5c7-fd6a-409a-bb8c-350918218f7c.jpg)
> 
> BTW, you can use `zine` to preview the magazine in advance on your local computer, see https://rustmagazine.org/contribution/

Thanks. I have fixed it.

Folyd reviewed (commented):
The subject is `"40x Faster! We rewrote our project with Rust"`, but it isn't an article about why and how to achieve 40x faster actually, instead an article `How do we feel after rewriting our project in Rust`, so I recommend giving a better title for this article.  

What do you think? @LCrossman @dmosc @macko76

LCrossman commented on content/issue-1/2023-kcl-rewrite-with-rust.md:
```diff
@@ -0,0 +1,139 @@
+
+## Development efficiency
+
+The development efficiency of Rust can be described as "restraining first and then improving". At the beginning of the handwritten project, if the team members have not been exposed to the concept of functional programming and related programming habits, the development speed will be significantly slower than Python, Go, Java and other languages. However, once they become familiar with the common ...
```
I don't understand what that phrase means here, nor "the handwritten project" to be honest

Peefy replied:
> I don't understand what that phrase means here, nor "the handwritten project" to be honest

fixed.

Peefy replied:
> It would be great to hear more about the 40 X - how did you benchmark it at 40 X and what language was it in before? Did it take a long time to rewrite it? Were there some particular steps that were bottleneck processes like maybe some iteration process that was taking a long time to execute in the the first language? Why did you choose Rust?

Thank you for your feedback. In the article, I explained that before using Rust, we used Python to write the entire project. Then we do a benchmark on virtual machine programs for Rust, Go, and Python, and compared the time cost. Finally, after comprehensive consideration, we chose Rust and spent several months rewriting the entire project. In addition, due to article length constraints, more content will be discussed in later articles.

Folyd commented:
Hi, @Peefy, could you please improve the English article? Otherwise, I would merge this. Sorry.

Folyd commented:
Maybe you can use ChatGPT to improve the article 👀

Peefy commented:
> Maybe you can use ChatGPT to improve the article 👀

Thank you for your suggestion. After I completely modify and polish the article, I will ping you to review.

Folyd commented:
We have released the first issue, so this article needs to be put off to issue 2.

Peefy commented:
> We have released the first issue, so this article needs to be put off to issue 2.

Hi @Folyd, I have polished the English content of the KCL article and moved it to issue-2. Thank you for your review.

Peefy commented:
> Hi @Folyd, I have polished the English content of the KCL article and moved it to issue-2. Thank you for your review.

Hi @Folyd, I have polished the English content of the KCL article again and moved it to issue-3. Thank you for your review.
